pkg/obfuscator/ip.go

```diff
@@ -32,29 +33,50 @@ var (
 )
 
 type ipGenerator struct {
-	template   string
-	obfuscated string
-	count      int
+	replacements map[string]string
+	template     string
+	obfuscated   string
+	count        int
+	lock         sync.Mutex
 }
 
-func (g *ipGenerator) generateConsistent() string {
+func (g *ipGenerator) consistent(ip string) string {
+	g.lock.Lock()
+	defer g.lock.Unlock()
+
+	if r, ok := g.replacements[ip]; ok {
+		return r
+	}
+
 	g.count++
 	if g.count > maximumSupportedObfuscations {
 		klog.Exitf("maximum number of ip obfuscations exceeded: %d", maximumSupportedObfuscations)
 	}
-	return fmt.Sprintf(g.template, g.count)
+	r := fmt.Sprintf(g.template, g.count)
+	g.replacements[ip] = r
+	return r
 }
 
-func (g *ipGenerator) static() string {
+func (g *ipGenerator) static(ip string) string {
+	g.replacements[ip] = g.obfuscated
```

Comment:
I think you also need to put this under the lock - what if the hashmap underneath rehashes the entries?

Reply:
Done